### PR TITLE
Number of levels for stedc tuned for CDNA2

### PR DESCRIPTION
**Pull request by afanfa**
Major improvements observed for CDNA2. Fair improvements on other architectures

### CODE REVIEW

File: library/src/auxiliary/rocauxiliary_stedc.hpp [hunk 1]
```diff
@@ -579,21 +579,39 @@ __host__ __device__ inline rocblas_int stedc_num_levels(const rocblas_int n)
...
+        else
+          {
+            levels = 6;
+          }
+      }
     else
-    {
-        levels = n / 8;
-        levels = floor(log(levels) / log(2));
-    }
```
Comment (cgmb) [inline library/src/auxiliary/rocauxiliary_stedc.hpp]: Ah. Nice to see this gone.

**Review by cgmb — COMMENTED** [review]
Please run
```
./scripts/get-clang-format $(git rev-parse --git-path hooks)
./scripts/reformat-files library/src/auxiliary/rocauxiliary_stedc.hpp
```
to apply the necessary formatting to fix the static analysis job on the CI.

File: library/src/auxiliary/rocauxiliary_stedc.hpp (hunk 1 above)
Comment (afanfa) [reply library/src/auxiliary/rocauxiliary_stedc.hpp]: Thanks for the quick review. Juan suggested some more appropriate testing of this routine. I am going to run a new tuning pass and update the PR with the new findings.

**Comment by afanfa** [comment]
I have tested on gfx1030 the entire testsuite and all tests passed. I can also confirm that the performance improvements are observable in various apps. @jzuniga-amd is this ok to merge?

**Review by jzuniga-amd — APPROVED** [review]
LGTM. Thanks Alessandro!

**Review by cgmb — APPROVED** [review]

**Pull request merged**